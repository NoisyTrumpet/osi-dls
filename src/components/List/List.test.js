import React from 'react';
import { render } from '@testing-library/react';
import {
  toBeInTheDocument,
  toContainHTML,
} from '@testing-library/jest-dom/matchers';
import { List } from './';

expect.extend({ toBeInTheDocument, toContainHTML });

describe('<List />', () => {
  test('should renders without props ', () => {
    const list = (
      <List>
        <List.Item>
          <span>Text</span>
        </List.Item>
        <List.Item>
          <span>Text</span>
        </List.Item>
        <List.Item>
          <span>Text</span>
        </List.Item>
      </List>
    );

    const { container } = render(list);
    expect(container.firstChild).toBeInTheDocument();
    expect(container.firstChild).not.toBeNull();
    expect(container).toMatchSnapshot();
  });

  test('should contain list of ordered items ', () => {
    const list = (
      <List ordered>
        <List.Item>
          <span>Text</span>
        </List.Item>
        <List.Item>
          <span>Text</span>
        </List.Item>
        <List.Item>
          <span>Text</span>
        </List.Item>
      </List>
    );

    const { container } = render(list);
    expect(container.firstChild).toContainHTML('<ol class="">');
    expect(container).toMatchSnapshot();
  });
});
