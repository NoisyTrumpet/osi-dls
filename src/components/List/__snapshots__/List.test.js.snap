// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should contain list of ordered items  1`] = `
<div>
  <ol
    class=""
  >
    <li
      class="list-item"
    >
      <span>
        Text
      </span>
    </li>
    <li
      class="list-item"
    >
      <span>
        Text
      </span>
    </li>
    <li
      class="list-item"
    >
      <span>
        Text
      </span>
    </li>
  </ol>
</div>
`;

exports[`<List /> should renders without props  1`] = `
<div>
  <ul
    class=""
  >
    <li
      class="list-item"
    >
      <span>
        Text
      </span>
    </li>
    <li
      class="list-item"
    >
      <span>
        Text
      </span>
    </li>
    <li
      class="list-item"
    >
      <span>
        Text
      </span>
    </li>
  </ul>
</div>
`;
